examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,579 @@
+  az_span signatureAzSpan = AZ_SPAN_FROM_BUFFER(signature);
+  uint8_t encodedSignedSignature[BUFFER_LENGTH_SAS_ENCODED_SIGNED_SIGNATURE] = {0};
+  size_t encodedSignedSignatureLength = 0;
+
+  // Get the signature. It will be signed later with the decoded device key.
+  // To change the sas token duration, see IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES in iot_configs.h
+  sasTokenDuration = getSASTokenExpirationTime(IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES);
+  result = az_iot_hub_client_sas_get_signature(
+      &azIoTHubClient, sasTokenDuration, signatureAzSpan, &signatureAzSpan);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the signature for SAS Token. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  // Sign and encode the signature (b64 encoded, HMAC-SHA256 signing).
+  // Uses the decoded device key.
+  generateSASBase64EncodedSignedSignature(
+      az_span_ptr(signatureAzSpan), az_span_size(signatureAzSpan),
+      encodedSignedSignature, sizeof(encodedSignedSignature), &encodedSignedSignatureLength);
+
+  // Get the MQTT password (SAS Token) from the base64 encoded, HMAC signed bytes.
+  az_span encodedSignedSignatureAzSpan = az_span_create(encodedSignedSignature, 
+                                                        encodedSignedSignatureLength);
+  result = az_iot_hub_client_sas_get_password(
+      &azIoTHubClient, sasTokenDuration, encodedSignedSignatureAzSpan, AZ_SPAN_EMPTY,
+      mqttPassword, sizeof(mqttPassword), NULL);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the MQTT password. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+}
+
+/*
+ * generateSASBase64EncodedSignedSignature:
+ * Sign and encode a signature. It is signed using the provided device key.
+ * The process is:
+ *    1. Decode the encoded device key.
+ *    2. Sign the signature with the decoded device key.
+ *    3. Encode the signed signature.
+ */
+static void generateSASBase64EncodedSignedSignature(
+    uint8_t const* sasSignature, size_t const sasSignatureSize,
+    uint8_t* encodedSignedSignature, size_t encodedSignedSignatureSize,
+    size_t* encodedSignedSignatureLength) 
+{
+  int result;
+  unsigned char sasDecodedKey[BUFFER_LENGTH_SAS] = {0};
+  size_t sasDecodedKeyLength = 0;
+  unsigned char sasHMAC256SignedSignature[BUFFER_LENGTH_SAS] = {0};
+  mbedtls_md_context_t ctx;
+  mbedtls_md_type_t md_type = MBEDTLS_MD_SHA256;
+
+  // Decode the SAS base64 encoded device key to use for HMAC signing.
+  result = mbedtls_base64_decode(
+      sasDecodedKey, sizeof(sasDecodedKey), &sasDecodedKeyLength,
+      (const unsigned char*)IOT_CONFIG_DEVICE_KEY, sizeof(IOT_CONFIG_DEVICE_KEY) - 1);
+  if (result != 0) 
+  {
+    logString = "mbedtls_base64_decode failed. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  // HMAC-SHA256 sign the signature with the decoded device key.
+  mbedtls_md_init(&ctx);

```

Comment:
The ATECC608A has HMAC crypto, you should be using that and not bringing the mbedtls lib to do this.

Also the device key should be be securely stored in the ATECC608A register.

Please see the microchip example for this.
https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx/blob/041e26dd4e619f099e14d167b8c9ad1a20deecb4/AzureIotPnpDps.X/mqtt_packetPopulation/mqtt_iothub_packetPopulate.c#L170

Reply:
Hey Eric, thanks for the feedback.  Unfortunately, we are behind on other work and need to shift focus from these samples.   We'd be happy to mention the ATECC608A in the Readme.

Reply:
Is there a list of what still needs to be completed?

Reply:
Working on implementing this on a second iteration